### ISSUE TITLE
Rule Proposal: Writing variable without reading

### ISSUE THREAD

**Issue by jens-duttke** [eslint/eslint#13862]
**Please describe what the rule should do:**

The rule should warn if a variable is overwritten without reading after the previous write.

**What new ECMAScript feature does this rule relate to?**

That issue can occur with every ECMAScript version.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Here, a warning should be throws, because neither `3` nor `4` is used.
```js
var foo = 3;
foo = 4;
foo = 5;
console.log(foo);
```

This is an issue I had before by refactoring some code (I forgot to wrap `foo = 2` in an `else` block):
```js
let foo;

if (Math.random() < 0.5) {
	foo = 1;
}

foo = 2;

console.log(foo);
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It fixes a common issue, which may cause unnecessary performance issues (by setting variables to unused values) or bugs.

For sure, it could be implemented in a plugin, but which one would be the best here? I assume the implementation is a bit more complex, so I'm not able to do it by my own, and I don't know where this problem is best kept.

**Are you willing to submit a pull request to implement this rule?**

No

**Comment by anikethsaha** [comment]
Thanks for the proposal. 
This is an interesting proposal but I think there may be some false cases too (those can be fixed with documenting them as a known limitation).
I think this wont work properly with objects , arrays etc.
